crates/subspace-farmer/src/single_disk_plot.rs

```diff
@@ -648,8 +653,18 @@ impl SingleDiskPlot {
                                     );
                                     return;
                                 }
-                                let piece_index = sector_id
-                                    .derive_piece_index(piece_offset as PieceIndex, total_pieces);
+                                let piece_index = match sector_id
```

Comment:
This file is extremely complex and hard to read. Consider splitting it into several modules, components, or functions, good candidates would be unsafe mmap operations, farming, plotting, single_disk_plot_info, metadata, etc. For example, the future DSN integration with plotting would be easier by having a separate plotting module.

Reply:
Agree, hopefully going through commits incrementally helps understanding how this was built. There are basically 2 background threads: one doing initial plotting and another doing farming. I'll move those into separate modules with upcoming PRs. Both plotting and farming actually share some code and the same code is also copy-pasted into testing utilities for light client and decoupled execution, all that has to be abstracted nicely, but the goal here was to get things going sooner and for that I decided to go with one file for now and moderate code duplication before patterns emerge. The rest is boilerplate for the most part without significant business logic.